### PR TITLE
Consul: allow clustering with nodes in warning status

### PR DESCRIPTION
**Pull request by ValFadeev**
**moved from https://github.com/aweber/rabbitmq-autocluster/pull/140**

Consul agent may be configured with additional health checks, besides the TTL one created by the plugin. Some of these checks may expect a converged cluster. If any of those additional checks is not passing the corresponding nodes will be excluded from the list for clustering which is not always the desired situation.

This change aims to provide flexibility to define additional health checks while avoiding interference with auto clustering. This is consistent with Consul's [`only-passing`](https://www.consul.io/docs/agent/options.html#only_passing) option which determines whether nodes with warning status are excluded from the DNS response.

### CODE REVIEW

**Comment by michaelklishin** [comment]
@ValFadeev thank you!

File: include/autocluster.hrl [hunk 1]
```diff
@@ -28,6 +28,7 @@
          {config, cluster_name,          "CLUSTER_NAME",           "undefined",  string,  false}, %% Consul && etcd
 
          {config, consul_acl_token,      "CONSUL_ACL_TOKEN",       "undefined",  string,  false}, %% Consul
+         {config, consul_allow_warn,     "CONSUL_ALLOW_WARN",      false,        atom,    false},
```
Comment (michaelklishin) [inline include/autocluster.hrl]: I'd like to find a more descriptive name. `CONSUL_TOLERATE_WARN` or `CONSUL_ALLOW_WARN` seem too short to me. It's not clear what are we allowing here.

I'd go with something like `CONSUL_INCLUDE_NODES_WITH_WARNINGS`. While longer, it is more or less to the point. WDYT?

Comment (ValFadeev) [reply include/autocluster.hrl]: @michaelklishin agree, in hindsight the current name may also cause confusion with `CLEANUP_WARN_ONLY` option. Happy to go with the suggested name, as can't think of anything shorter to convey the meaning

**Review by dcorbacho — APPROVED** [review]
Thanks @ValFadeev! It looks good, only added an additional test case.

**Pull request merged**